### PR TITLE
Adds closeOnDateSelection() to date time picker

### PR DESCRIPTION
**Pull request by johncarter-**
👆

### CODE REVIEW

File: packages/forms/src/Components/DateTimePicker.php [hunk 1]
```diff
@@ -30,6 +30,8 @@ class DateTimePicker extends Field
 
     protected bool | Closure $isWithoutTime = false;
 
+    protected bool | Closure $closeOnDateSelection = false;
```
Comment (zepfietje) [inline packages/forms/src/Components/DateTimePicker.php:33]:
```suggestion
    protected bool | Closure $shouldCloseOnDateSelection = false;
```

File: packages/forms/src/Components/DateTimePicker.php [hunk 2]
```diff
@@ -182,6 +184,13 @@ public function withoutTime(bool | Closure $condition = true): static
         return $this;
     }
 
+    public function closeOnDateSelection(bool | Closure $condition = true): static
+    {
+        $this->closeOnDateSelection = $condition;
```
Comment (zepfietje) [inline packages/forms/src/Components/DateTimePicker.php:189]:
```suggestion
        $this->shouldCloseOnDateSelection = $condition;
```

File: packages/forms/src/Components/DateTimePicker.php [hunk 3]
```diff
@@ -273,6 +282,11 @@ public function hasTime(): bool
         return ! $this->isWithoutTime;
     }
 
+    public function shouldCloseOnDateSelection(): bool
+    {
+        return $this->closeOnDateSelection;
```
Comment (zepfietje) [inline packages/forms/src/Components/DateTimePicker.php:287]:
```suggestion
        return $this->evaluate($this->shouldCloseOnDateSelection);
```

**Review by zepfietje — CHANGES_REQUESTED** [review]

**Comment by zepfietje** [comment]
I've added some comments to update naming to make it similar to existing code.

`closeOnDateSelection` still feels a bit long to me though. It is clear, but would something like `closeOnSelect` also suffice? @danharrin

**Comment by johncarter-** [comment]
I'll get those changes made asap. Just for reference: https://github.com/filamentphp/filament/discussions/3060#discussioncomment-3108418

**Comment by zepfietje** [comment]
> I'll get those changes made asap. Just for reference: [#3060 (comment)](https://github.com/filamentphp/filament/discussions/3060#discussioncomment-3108418)

Makes sense!

**Comment by danharrin** [comment]
`closeOnDateSelection()` is appropriate here because it should not close when we select a time, in which case `closeOnSelection()` is inappropriate

**Comment by zepfietje** [comment]
@johncarter-, I've updated the implementation to make sure the panel also closes when selecting dates using the keyboard. This is ready to be merged if @danharrin approves.

**Comment by johncarter-** [comment]
Thank you, mate.

**Pull request merged**